api/lib/reducers/cosmosV2-reducers.js

```diff
@@ -44,44 +44,56 @@ function setTransactionSuccess(transaction, index) {
     : false
 }
 
-function sendDetailsReducer(message, reducers) {
+function sendDetailsReducer(message, reducers, network) {
+  const coinLookup = network.coinLookup.find(
+    ({ chainDenom }) => chainDenom === message.amount[0].denom
+  )
   return {
     from: [message.from_address],
     to: [message.to_address],
-    amount: reducers.coinReducer(message.amount[0])
+    amount: reducers.coinReducer(message.amount[0], coinLookup)
   }
 }
 
-function stakeDetailsReducer(message, reducers) {
+function stakeDetailsReducer(message, reducers, network) {
+  const coinLookup = network.coinLookup.find(
+    ({ chainDenom }) => chainDenom === message.amount.denom
+  )
   return {
     to: [message.validator_address],
-    amount: reducers.coinReducer(message.amount)
+    amount: reducers.coinReducer(message.amount, coinLookup)
   }
 }
 
-function restakeDetailsReducer(message, reducers) {
+function restakeDetailsReducer(message, reducers, network) {
+  const coinLookup = network.coinLookup.find(
+    ({ chainDenom }) => chainDenom === message.amount.denom
+  )
   return {
     from: [message.validator_src_address],
     to: [message.validator_dst_address],
-    amount: reducers.coinReducer(message.amount)
+    amount: reducers.coinReducer(message.amount, coinLookup)
   }
 }
 
-function unstakeDetailsReducer(message, reducers) {
+function unstakeDetailsReducer(message, reducers, network) {
+  const coinLookup = network.coinLookup.find(
```

Comment:
this snippet is repeated a lot. let's refactor and create a function `getCoinLookup(network, denom)`